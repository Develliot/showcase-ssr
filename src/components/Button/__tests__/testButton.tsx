import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Button } from '../';

it('renders correctly with children', () => {
    const wrapper: ShallowWrapper = shallow(
        <Button>
            <div>Hello</div>
            <div>World</div>
        </Button>
    );
    expect(wrapper).toMatchSnapshot();
});

it('renders correctly with a different color', () => {
    const wrapper: ShallowWrapper = shallow(
        <Button color='pink'>
            <div>Hello</div>
        </Button>
    );
    expect(wrapper).toMatchSnapshot();
});

it('works with a click handler', () => {
    const handleClick = jest.fn();
    const wrapper: ShallowWrapper = shallow(
        <Button color='pink' onClick={handleClick}>
            <div>Hello</div>
        </Button>
    );
    wrapper.simulate('click');
    expect(handleClick).toHaveBeenCalledTimes(1);
});
