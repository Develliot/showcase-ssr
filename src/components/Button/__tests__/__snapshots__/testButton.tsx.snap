// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a different color 1`] = `
<styles__ButtonStyled
  color="pink"
  onClick={[Function]}
>
  <div>
    Hello
  </div>
</styles__ButtonStyled>
`;

exports[`renders correctly with children 1`] = `
<styles__ButtonStyled
  color="blue"
  onClick={[Function]}
>
  <div>
    Hello
  </div>
  <div>
    World
  </div>
</styles__ButtonStyled>
`;
